### DISCUSSION TITLE
How to Use the FormData API

### DISCUSSION THREAD

**Discussion by thecodeclimber** [react-hook-form/react-hook-form#2912] [General]
I had a situation where I needed to send data using the FormData API.  I had some trouble getting this to work and wanted to share the solution I eventually came up with in case it can help anyone else.  Here it is:

```
const onSubmit = (fData) => {
  const data = new FormData()

  for (const key in fData) {
    if (key === 'field') {
      data.append(key, fData[key][1])
    } else {
      data.append(key, fData[key])
    }
  }
  ...
```

Hope that helps.

**Comment by bluebill1049** [comment]
Thanks for sharing solutions.

**Comment by psaikali** [comment]
Helped me today, thanks @thecodeclimber :)

**Comment by lucky210788** [comment]
Thank you

**Comment by CIDBerlin** [comment]
Thanks, bro <3

**Comment by andreknieriem** [comment]
Thanks!

**Comment by bstevary** [comment]
Thanks, solved a one week nightmare today
